Running make fix after make verify fails

### Bug report criteria

- [X] This bug report is not security related, security issues should be disclosed privately via [etcd maintainers](mailto:etcd-maintainers@googlegroups.com).
- [X] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [X] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [X] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

When running `make fix` after `make verify` fails. This could impact the Prow job that runs those two commands one after the other. Refer to: https://github.com/kubernetes/test-infra/blob/b5839483b26c047ef83152e4ce2e47442954ce53/config/jobs/etcd/etcd-presubmits.yaml#L67-L71

The error running `make fix` after running `make verify` without having yamllint installed, is the following:

```
make fix
which: no yamllint in (/home/ivan/.local/share/asdf/installs/golang/1.22.2/packages/bin:/home/ivan/bin:/home/ivan/.local/share/asdf/installs/golang/1.22.2/packages/bin:/home/ivan/bin:/home/ivan/.local/share/asdf/shims:/opt/asdf-vm/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl)
./scripts/updatebom.sh
generating bill-of-materials.json
% (cd tools/mod && 'go' 'install' 'github.com/appscodelabs/license-bill-of-materials')
% '/home/ivan/.local/share/asdf/installs/golang/1.22.2/packages/bin/license-bill-of-materials' '--override-file' './bill-of-materials.override.json' 'go.etcd.io/etcd/api/v3/...' 'go.etcd.io/etcd/pkg/v3/...' 'go.etcd.io/etcd/client/pkg/v3/...' 'go.etcd.io/etcd/client/v2/...' 'go.etcd.io/etcd/client/v3/...' 'go.etcd.io/etcd/server/v3/...' 'go.etcd.io/etcd/etcdutl/v3/...' 'go.etcd.io/etcd/etcdctl/v3/...' 'go.etcd.io/etcd/tests/v3/...' 'go.etcd.io/etcd/v3/...'
FAIL: (code:1):
  % '/home/ivan/.local/share/asdf/installs/golang/1.22.2/packages/bin/license-bill-of-materials' '--override-file' './bill-of-materials.override.json' 'go.etcd.io/etcd/api/v3/...' 'go.etcd.io/etcd/pkg/v3/...' 'go.etcd.io/etcd/client/pkg/v3/...' 'go.etcd.io/etcd/client/v2/...' 'go.etcd.io/etcd/client/v3/...' 'go.etcd.io/etcd/server/v3/...' 'go.etcd.io/etcd/etcdutl/v3/...' 'go.etcd.io/etcd/etcdctl/v3/...' 'go.etcd.io/etcd/tests/v3/...' 'go.etcd.io/etcd/v3/...'
FAIL: bom refreshing failed
stderr: 2024/05/04 09:44:04 could not list go.etcd.io/etcd/api/v3/... go.etcd.io/etcd/pkg/v3/... go.etcd.io/etcd/client/pkg/v3/... go.etcd.io/etcd/client/v2/... go.etcd.io/etcd/client/v3/... go.etcd.io/etcd/server/v3/... go.etcd.io/etcd/etcdutl/v3/... go.etcd.io/etcd/etcdctl/v3/... go.etcd.io/etcd/tests/v3/... go.etcd.io/etcd/v3/... dependencies: 'go list -f {{range .Deps}}{{.}}|{{end}} warning: ignoring symlink /home/ivan/Code/Personal/etcd/etcd/bin/python/lib64 warning: ignoring symlink /home/ivan/Code/Personal/etcd/etcd/bin/python/lib64 go.etcd.io/etcd/api/v3/authpb go.etcd.io/etcd/api/v3/etcdserverpb go.etcd.io/etcd/api/v3/etcdserverpb/gw go.etcd.io/etcd/api/v3/membershippb go.etcd.io/etcd/api/v3/mvccpb go.etcd.io/etcd/api/v3/v3rpc/rpctypes go.etcd.io/etcd/api/v3/version go.etcd.io/etcd/api/v3/versionpb go.etcd.io/etcd/pkg/v3/adt go.etcd.io/etcd/pkg/v3/cobrautl go.etcd.io/etcd/pkg/v3/contention go.etcd.io/etcd/pkg/v3/cpuutil go.etcd.io/etcd/pkg/v3/crc go.etcd.io/etcd/pkg/v3/debugutil go.etcd.io/etcd/pkg/v3/expect go.etcd.io/etcd/pkg/v3/flags go.etcd.io/etcd/pkg/v3/grpctesting go.etcd.io/etcd/pkg/v3/httputil go.etcd.io/etcd/pkg/v3/idutil go.etcd.io/etcd/pkg/v3/ioutil go.etcd.io/etcd/pkg/v3/netutil go.etcd.io/etcd/pkg/v3/notify go.etcd.io/etcd/pkg/v3/osutil go.etcd.io/etcd/pkg/v3/pbutil go.etcd.io/etcd/pkg/v3/proxy go.etcd.io/etcd/pkg/v3/report go.etcd.io/etcd/pkg/v3/runtime go.etcd.io/etcd/pkg/v3/schedule go.etcd.io/etcd/pkg/v3/stringutil go.etcd.io/etcd/pkg/v3/traceutil go.etcd.io/etcd/pkg/v3/wait go.etcd.io/etcd/client/pkg/v3/fileutil go.etcd.io/etcd/client/pkg/v3/logutil go.etcd.io/etcd/client/pkg/v3/pathutil go.etcd.io/etcd/client/pkg/v3/srv go.etcd.io/etcd/client/pkg/v3/systemd go.etcd.io/etcd/client/pkg/v3/testutil go.etcd.io/etcd/client/pkg/v3/tlsutil go.etcd.io/etcd/client/pkg/v3/transport go.etcd.io/etcd/client/pkg/v3/types go.etcd.io/etcd/client/pkg/v3/verify go.etcd.io/etcd/client/v2 go.etcd.io/etcd/client/v3 go.etcd.io/etcd/client/v3/clientv3util go.etcd.io/etcd/client/v3/concurrency go.etcd.io/etcd/client/v3/credentials go.etcd.io/etcd/client/v3/experimental/recipes go.etcd.io/etcd/client/v3/internal/endpoint go.etcd.io/etcd/client/v3/internal/resolver go.etcd.io/etcd/client/v3/leasing go.etcd.io/etcd/client/v3/mirror go.etcd.io/etcd/client/v3/mock/mockserver go.etcd.io/etcd/client/v3/namespace go.etcd.io/etcd/client/v3/naming go.etcd.io/etcd/client/v3/naming/endpoints go.etcd.io/etcd/client/v3/naming/endpoints/internal go.etcd.io/etcd/client/v3/naming/resolver go.etcd.io/etcd/client/v3/ordering go.etcd.io/etcd/client/v3/snapshot go.etcd.io/etcd/client/v3/yaml go.etcd.io/etcd/server/v3 go.etcd.io/etcd/server/v3/auth go.etcd.io/etcd/server/v3/config go.etcd.io/etcd/server/v3/embed go.etcd.io/etcd/server/v3/etcdmain go.etcd.io/etcd/server/v3/etcdserver go.etcd.io/etcd/server/v3/etcdserver/api go.etcd.io/etcd/server/v3/etcdserver/api/etcdhttp go.etcd.io/etcd/server/v3/etcdserver/api/etcdhttp/types go.etcd.io/etcd/server/v3/etcdserver/api/membership go.etcd.io/etcd/server/v3/etcdserver/api/rafthttp go.etcd.io/etcd/server/v3/etcdserver/api/snap go.etcd.io/etcd/server/v3/etcdserver/api/snap/snappb go.etcd.io/etcd/server/v3/etcdserver/api/v2discovery go.etcd.io/etcd/server/v3/etcdserver/api/v2error go.etcd.io/etcd/server/v3/etcdserver/api/v2stats go.etcd.io/etcd/server/v3/etcdserver/api/v2store go.etcd.io/etcd/server/v3/etcdserver/api/v3alarm go.etcd.io/etcd/server/v3/etcdserver/api/v3client go.etcd.io/etcd/server/v3/etcdserver/api/v3compactor go.etcd.io/etcd/server/v3/etcdserver/api/v3discovery go.etcd.io/etcd/server/v3/etcdserver/api/v3election go.etcd.io/etcd/server/v3/etcdserver/api/v3election/v3electionpb go.etcd.io/etcd/server/v3/etcdserver/api/v3election/v3electionpb/gw go.etcd.io/etcd/server/v3/etcdserver/api/v3lock go.etcd.io/etcd/server/v3/etcdserver/api/v3lock/v3lockpb go.etcd.io/etcd/server/v3/etcdserver/api/v3lock/v3lockpb/gw go.etcd.io/etcd/server/v3/etcdserver/api/v3rpc go.etcd.io/etcd/server/v3/etcdserver/apply go.etcd.io/etcd/server/v3/etcdserver/cindex go.etcd.io/etcd/server/v3/etcdserver/errors go.etcd.io/etcd/server/v3/etcdserver/txn go.etcd.io/etcd/server/v3/etcdserver/version go.etcd.io/etcd/server/v3/lease go.etcd.io/etcd/server/v3/lease/leasehttp go.etcd.io/etcd/server/v3/lease/leasepb go.etcd.io/etcd/server/v3/mock/mockstorage go.etcd.io/etcd/server/v3/mock/mockstore go.etcd.io/etcd/server/v3/mock/mockwait go.etcd.io/etcd/server/v3/proxy/grpcproxy go.etcd.io/etcd/server/v3/proxy/grpcproxy/adapter go.etcd.io/etcd/server/v3/proxy/grpcproxy/cache go.etcd.io/etcd/server/v3/proxy/tcpproxy go.etcd.io/etcd/server/v3/storage go.etcd.io/etcd/server/v3/storage/backend go.etcd.io/etcd/server/v3/storage/backend/testing go.etcd.io/etcd/server/v3/storage/datadir go.etcd.io/etcd/server/v3/storage/mvcc go.etcd.io/etcd/server/v3/storage/mvcc/testutil go.etcd.io/etcd/server/v3/storage/schema go.etcd.io/etcd/server/v3/storage/wal go.etcd.io/etcd/server/v3/storage/wal/testing go.etcd.io/etcd/server/v3/storage/wal/walpb go.etcd.io/etcd/server/v3/verify go.etcd.io/etcd/etcdutl/v3 go.etcd.io/etcd/etcdutl/v3/etcdutl go.etcd.io/etcd/etcdutl/v3/snapshot go.etcd.io/etcd/etcdctl/v3 go.etcd.io/etcd/etcdctl/v3/ctlv3 go.etcd.io/etcd/etcdctl/v3/ctlv3/command go.etcd.io/etcd/tests/v3/common go.etcd.io/etcd/tests/v3/e2e go.etcd.io/etcd/tests/v3/framework go.etcd.io/etcd/tests/v3/framework/config go.etcd.io/etcd/tests/v3/framework/e2e go.etcd.io/etcd/tests/v3/framework/integration go.etcd.io/etcd/tests/v3/framework/interfaces go.etcd.io/etcd/tests/v3/framework/testutils go.etcd.io/etcd/tests/v3/framework/unit go.etcd.io/etcd/tests/v3/integration go.etcd.io/etcd/tests/v3/integration/clientv3 go.etcd.io/etcd/tests/v3/integration/clientv3/concurrency go.etcd.io/etcd/tests/v3/integration/clientv3/connectivity go.etcd.io/etcd/tests/v3/integration/clientv3/examples go.etcd.io/etcd/tests/v3/integration/clientv3/experimental/recipes go.etcd.io/etcd/tests/v3/integration/clientv3/lease go.etcd.io/etcd/tests/v3/integration/clientv3/naming go.etcd.io/etcd/tests/v3/integration/clientv3/snapshot go.etcd.io/etcd/tests/v3/integration/embed go.etcd.io/etcd/tests/v3/integration/proxy/grpcproxy go.etcd.io/etcd/tests/v3/integration/snapshot go.etcd.io/etcd/tests/v3/integration/v2store go.etcd.io/etcd/tests/v3/robustness go.etcd.io/etcd/tests/v3/robustness/failpoint go.etcd.io/etcd/tests/v3/robustness/identity go.etcd.io/etcd/tests/v3/robustness/model go.etcd.io/etcd/tests/v3/robustness/options go.etcd.io/etcd/tests/v3/robustness/report go.etcd.io/etcd/tests/v3/robustness/traffic go.etcd.io/etcd/tests/v3/robustness/validate go.etcd.io/etcd/v3 go.etcd.io/etcd/v3/contrib/lock/client go.etcd.io/etcd/v3/contrib/lock/storage go.etcd.io/etcd/v3/contrib/raftexample go.etcd.io/etcd/v3/tools/benchmark go.etcd.io/etcd/v3/tools/benchmark/cmd go.etcd.io/etcd/v3/tools/etcd-dump-db go.etcd.io/etcd/v3/tools/etcd-dump-logs go.etcd.io/etcd/v3/tools/etcd-dump-metrics go.etcd.io/etcd/v3/tools/local-tester/bridge go.etcd.io/etcd/v3/tools/proto-annotations go.etcd.io/etcd/v3/tools/proto-annotations/cmd' failed with:
stderr: malformed import path "warning: ignoring symlink /home/ivan/Code/Personal/etcd/etcd/bin/python/lib64": invalid char ':'
make: *** [Makefile:74: fix-bom] Error 2
```

This is due to the virtual environment generated to run `yamllint` introduced in b103d46ce7.

### What did you expect to happen?

It should run without issues.

### How can we reproduce it (as minimally and precisely as possible)?

Run `make verify fix`

### Anything else we need to know?

_No response_

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
# paste output here

$ etcdctl version
# paste output here
```

</details>


### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>


### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>


### Relevant log output

_No response_

--- a/Makefile
+++ b/Makefile
@@ -125,9 +125,11 @@ verify-genproto:
 verify-yamllint:
 ifeq (, $(shell which yamllint))
 	@echo "Installing yamllint..."
-	python3 -m venv bin/python
-	bin/python/bin/python3 -m pip install yamllint
-	./bin/python/bin/yamllint --config-file tools/.yamllint .
+	tmpdir=$$(mktemp -d); \
+	trap "rm -rf $$tmpdir" EXIT; \
+	python3 -m venv $$tmpdir; \
+	$$tmpdir/bin/python3 -m pip install yamllint; \
+	$$tmpdir/bin/yamllint --config-file tools/.yamllint .
 else
 	@echo "yamllint already installed..."
 	yamllint --config-file tools/.yamllint .

